src/historical-stipend-calculator.ts

```diff
@@ -0,0 +1,342 @@
+  const businessEntertainmentCost = BUSINESS_ENTERTAINMENT_PER_DAY * numberOfMealDays;
+  const localTransportCost = calculateLocalTransportCost(destination, numberOfMealDays, colFactor, BASE_LOCAL_TRANSPORT_PER_DAY);
+
+  // Parse ticket price, defaulting to 0 if not provided
+  const ticketPrice = record["Ticket Price (USD)"] ? parseFloat(record["Ticket Price (USD)"]) : 0;
+
+  // Total stipend is the sum of all expenses
+  const totalStipend = flightCost + lodgingCost + mealsCost + businessEntertainmentCost + localTransportCost + ticketPrice;
+
+  // Determine if international travel
+  const isInternational = ORIGIN.toLowerCase().includes("korea") && !destination.toLowerCase().includes("korea");
+
+  // Add internet/data allowance
+  const internetDataAllowance = isInternational ? 25 : 0;
+
+  // Add incidentals allowance
+  const incidentalsAllowance = numberOfMealDays * 20;
+
+  // Add to total stipend
+  const updatedTotalStipend = totalStipend + internetDataAllowance + incidentalsAllowance;
+
+  const result: StipendBreakdown = {
+    conference: record["Event Name"],
+    location: destination,
+    conference_start: dates.start,
+    conference_end: dates.end,
+    flight_departure: flightDates.outbound,
+    flight_return: flightDates.return,
+    distance_km: distanceKm,
+    flight_cost: parseFloat(flightCost.toFixed(2)),
+    lodging_cost: parseFloat(lodgingCost.toFixed(2)),
+    basic_meals_cost: parseFloat(mealsCost.toFixed(2)),
+    business_entertainment_cost: parseFloat(businessEntertainmentCost.toFixed(2)),
+    local_transport_cost: parseFloat(localTransportCost.toFixed(2)),
+    ticket_price: ticketPrice,
+    internet_data_allowance: internetDataAllowance,
+    incidentals_allowance: incidentalsAllowance,
+    total_stipend: parseFloat(updatedTotalStipend.toFixed(2)),
+    meals_cost: parseFloat(mealsCost.toFixed(2)),
+  };
+
+  stipendCache.set(cacheKey, result);
+  return result;
+}
+
+async function main() {
+  console.log("Starting historical travel stipend calculation...");
+
+  // Create a temporary CSV file from the provided data
+  const historicalData = `Event Name,Dates,Ticket Price (USD)
+"Asia Blockchain Summit 2024 - Taipei","August 6–8, 2024",0
+"Korea Blockchain Week 2024 - Seoul","September 1–7, 2024",0
+"TOKEN2049 Singapore 2024","September 18–19, 2024",599
+"GitHub Universe 2024 - San Francisco","October 29–30, 2024",1300
+"Devcon 7 - Bangkok","November 12–15, 2024",284.76
+"Consensus Hong Kong 2025","February 18–20, 2025",500`;
+
+  const tempCsvPath = path.join("fixtures", "historical-conferences.csv");
+  fs.writeFileSync(tempCsvPath, historicalData);
+
+  // Read conference data from temporary CSV file
+  console.log("Reading historical conference data...");
+  const fileContent = fs.readFileSync(tempCsvPath, "utf-8");
+  const records: HistoricalConference[] = parse(fileContent, {
+    columns: true,
+    skip_empty_lines: true,
+  });
+  console.log(`Loaded ${records.length} conference records`);
```

Comment:
try catch csv parsing

```suggestion
// Read conference data from temporary CSV file
  console.log("Reading historical conference data...");
  let records: HistoricalConference[] = [];
  try {
    const fileContent = fs.readFileSync(tempCsvPath, "utf-8");
    records = parse(fileContent, {
      columns: true,
      skip_empty_lines: true,
    });
    console.log(`Loaded ${records.length} conference records`);
  } catch (error) {
    console.error("Error reading or parsing CSV file:", error);
    process.exit(1);
  }
  ```

Reply:
No the program should fail if it can't parse the csv that's the whole point of it.